src/coreComponents/linearAlgebra/interfaces/hypre/mgrStrategies/MultiphasePoromechanics.hpp

```diff
@@ -71,17 +71,17 @@ class MultiphasePoromechanics : public MGRStrategyBase< 3 >
 
     setupLabels();
 
-    m_levelFRelaxMethod[0] = 2; // AMG V-cycle
-    m_levelFRelaxMethod[1] = 0; // Jacobi
-    m_levelFRelaxMethod[2] = 0; // Jacobi
+    m_levelFRelaxMethod[0] = toUnderlying( hypre::MGRFRelaxationMethod::amgVCycle );
+    m_levelFRelaxMethod[1] = toUnderlying( hypre::MGRFRelaxationMethod::singleLevel ); //default, i.e. Jacobi (to be confirmed)
+    m_levelFRelaxMethod[2] = toUnderlying( hypre::MGRFRelaxationMethod::singleLevel ); //default, i.e. Jacobi (to be confirmed)
 
     // Specified for the first level and default for the remaining
-    m_levelInterpType[0] = 2;       // diagonal scaling (Jacobi)
-    m_levelCoarseGridMethod[0] = 1; // diagonal sparsification
-    m_levelInterpType[1] = 2;       // diagonal scaling (Jacobi)
-    m_levelCoarseGridMethod[1] = 0; // standard Galerkin
-    m_levelInterpType[2] = 2;       // diagonal scaling (Jacobi)
-    m_levelCoarseGridMethod[2] = 0; // standard Galerkin
+    m_levelInterpType[0] = toUnderlying( hypre::MGRLevelInterpolationType::jacobi );
+    m_levelCoarseGridMethod[0] =toUnderlying( hypre::MGRLevelCoarseGridMethod::nonGalerkin );
+    m_levelInterpType[1] = toUnderlying( hypre::MGRLevelInterpolationType::jacobi );
+    m_levelCoarseGridMethod[1] = toUnderlying( hypre::MGRLevelCoarseGridMethod::galerkin );
+    m_levelInterpType[2] = toUnderlying( hypre::MGRLevelInterpolationType::jacobi );
+    m_levelCoarseGridMethod[2] = toUnderlying( hypre::MGRLevelCoarseGridMethod::galerkin );
```

Comment:
Would it be feasible to change the declaration of all these variables in `MGRStrategyBase` to actually be `enum`s, and only call `toUnderlying` when passing values into _hypre_ calls? Thus we can enforce stronger typing on our side and avoid potential errors.

This might not work if at any point a _hypre_ call requires us to pass a **pointer** to `HYPRE_Int`.

Reply:
Done. Following your suggestion, the **pointer** to `HYPRE_Int` has been addressed using `reinterpret_cast`, (see function [toUnderlyingPtr](https://github.com/GEOSX/GEOSX/blob/e52301b7dd5a60da6d65bde1bc74b29576f08069/src/coreComponents/codingUtilities/Utilities.hpp#L200-L210)).